pkg/proxy/proxy_test.go

```diff
@@ -0,0 +1,93 @@
+package proxy
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestRequestValidation(t *testing.T) {
+	tests := []struct {
+		name       string
+		method     string
+		subnet     string
+		hostname   string
+		wantStatus int
+		wantErr    bool
+	}{
+		{
+			name:       "get https same subnet",
+			method:     http.MethodGet,
+			subnet:     "127.0.0.1/24",
+			hostname:   "https://127.0.0.2:123",
+			wantStatus: http.StatusMethodNotAllowed,
+			wantErr:    true,
+		},
+		{
+			name:       "get https different subnet",
+			method:     http.MethodGet,
+			subnet:     "127.0.0.1/24",
+			hostname:   "https://10.0.0.2:123",
+			wantStatus: http.StatusMethodNotAllowed,
+			wantErr:    true,
+		},
```

Comment:
Is there a reason we have or need this test case?  It's going to always result in the same as the one above `"get https same subnet"` since it's using the `http.MethodGet`.

Reply:
The result is going to be the same, true. I usually try to be on the more cautious side testing-wise, but I agree that this one might be superfluous.